src/shared/components/common/markdown-textarea.tsx

```diff
@@ -346,61 +351,76 @@ export class MarkdownTextArea extends Component<
   }
 
   handleImageUpload(i: MarkdownTextArea, event: any) {
-    let file: any;
-    if (event.target) {
-      event.preventDefault();
-      file = event.target.files[0];
-    } else {
-      file = event;
+    i.setState({ imageLoading: true });
+    // TODO: unclear if this is needed
+    if (!event.target) {
+      i.uploadImage(i, event)
+      i.setState({ imageLoading: false });
+      return
+    } 
+    if (event.target.files.length > maxUploadImages) {
+        toast(i18n.t("too_many_images_upload", {
+          count: maxUploadImages,
+          formattedCount: numToSI(maxUploadImages),
+        }), "danger");
+        i.setState({ imageLoading: false });
+        return
     }
-
+    async function processFiles() {
+      await PromisePool
```

Comment:
Can probably just use Promise.all() on this, rather than pulling in a new dependency.

Reply:
cc @sam365724